superset/models/dashboard.py

```diff
@@ -67,17 +67,19 @@
 logger = logging.getLogger(__name__)
 
 
-def copy_dashboard(_mapper: Mapper, connection: Connection, target: Dashboard) -> None:
+def copy_dashboard(  # pylint: disable=unused-argument
+    _mapper: Mapper,
+    connection: Connection,
+    target: Dashboard,
+) -> None:
     dashboard_id = config["DASHBOARD_TEMPLATE_ID"]
     if dashboard_id is None:
         return
 
-    session_class = sessionmaker(autoflush=False)
-    session = session_class(bind=connection)
-    new_user = session.query(User).filter_by(id=target.id).first()
+    new_user = db.session.query(User).filter_by(id=target.id).first()
 
     # copy template dashboard to user
-    template = session.query(Dashboard).filter_by(id=int(dashboard_id)).first()
+    template = db.session.query(Dashboard).filter_by(id=int(dashboard_id)).first()
```

Comment:
Shouldn't we use the `connection` parameter here? According to the [event definition](https://docs.sqlalchemy.org/en/20/orm/events.html#sqlalchemy.orm.MapperEvents.after_insert):

`connection` – the `Connection` being used to emit INSERT statements for this instance. This provides a handle into the current transaction on the target database specific to this instance.

Reply:
@michael-s-molina I've reverted the PR back to my initial implementation which leverages the `connection` parameter. It states that it provides a handle into the current transaction, yet we're likely over committing, i.e., there should only be one commit per unit of work. I'm not sure what happens if you close the session without committing.  Maybe we using the connection wrongly.

Reply:
Regrettably I was running into [this](https://stackoverflow.com/questions/43164546/sawarning-usage-of-the-session-add-operation-is-not-currently-supported-wit) issue and had to revert my last commit. I think we likely will have to revisit how tagging is working at a later date.